packages/api-docs-builder/buildApiUtils.ts

```diff
@@ -9,9 +9,17 @@ import { replaceComponentLinks } from './utils/replaceUrl';
 import findPagesMarkdownNew from './utils/findPagesMarkdown';
 import { TypeScriptProject } from './utils/createTypeScriptProject';
 
-const systemComponents = fs
-  .readdirSync(path.resolve('packages', 'mui-system', 'src'))
-  .filter((pathname) => pathname.match(/^[A-Z][a-zA-Z]+$/));
+let systemComponents: string[] | undefined;
+// making the resolution lazy to avoid issues when importing something irrelevant from this file (i.e. `getSymbolDescription`)
+// the eager resolution results in errors when consuming externally (i.e. `mui-x`)
+function getSystemComponents() {
+  if (!systemComponents) {
```

Comment:
@siriwatknp Maybe you have any better idea of how to resolve the issue mentioned here https://github.com/mui/material-ui/pull/36291#discussion_r1114266061?